File: Java/프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.java
import java.util.*;

class Solution {
    public int[] solution(String[] keymap, String[] targets) {
        int[] answer = new int[targets.length];
        HashMap<Character, Integer> keyIndex = new HashMap<>();
        
        for (String key : keymap) {
            for (int i = 0; i < key.length(); i++) {
                Character keyItem = key.charAt(i);
                if (keyIndex.getOrDefault(keyItem, 101) > (i+1)) {
                    keyIndex.put(keyItem, i+1);
                }
            }
        }
        
        for (int i = 0; i < targets.length; i++) {
            String target = targets[i];
            int result = 0;
            for (int j = 0; j < target.length(); j++) {
                Character keyItem = target.charAt(j);
                int index = keyIndex.getOrDefault(keyItem, -1);
                if (index == -1) {
                    result = -1;
                    break;
                }
                result += index;
            }
            answer[i] = result;
        }
        return answer;
    }
}

File: Swift/프로그래머스/1/160586. 대충 만든 자판/대충 만든 자판.swift
import Foundation

func solution(_ keymap:[String], _ targets:[String]) -> [Int] {
    var keyDic: [String : Int] = [:]
    for keyMatch in keymap {
        for i in 0..<keyMatch.count {
            let index = keyMatch.index(keyMatch.startIndex, offsetBy: i)
            let keyChar = String(keyMatch[index])
            if let keyMin = keyDic[keyChar], keyMin <= i+1 {
                continue
            } else {
                keyDic[keyChar] = i+1
            }
        }
    }
    
    var answer: [Int] = []
    for target in targets {
        var result: Int = 0
        for i in 0..<target.count {
            let index = target.index(target.startIndex, offsetBy: i)
            let keyChar = String(target[index])
            if let keyMin = keyDic[keyChar] {
                result += keyMin
            } else {
                result = -1
                break
            }
        }
        answer.append(result)
    }
    
    return answer
}